zjit/src/codegen.rs

```diff
@@ -216,6 +215,8 @@ fn gen_function(cb: &mut CodeBlock, iseq: IseqPtr, function: &Function) -> Optio
                 let new_sp = asm.sub(NATIVE_STACK_PTR, jit.c_stack_bytes.into());
                 asm.mov(NATIVE_STACK_PTR, new_sp);
             }
+
+            gen_entry_params(&mut asm, iseq, function.block(BlockId(0)));
```

Comment:
We deliberately put this in `gen_entry()` instead of here so that JIT-to-JIT calls don't have to read arguments off of EP. When we optimize frame push further by removing eager spilling, this would stop working.

Could we adjust operands in `jit.opnds` for params if they point to memory based off of `NATIVE_STACK_PTR` instead? If it doesn't work, let's leave a TODO comment to optimize argument access on JIT-to-JIT calls.

Reply:
Wrote to the slots at entry point in [604c715](https://github.com/ruby/ruby/pull/13802/commits/604c71571c6ea1c70de061d36a6993c542e16a6e).